File: 프로그래머스/Level 2/스킬트리-ex.py
def solution(skill, skill_trees):
    possible_skill_tree_num = 0
    
    for skill_tree in skill_trees:
        skill_order = [s for s in skill_tree if s in skill]
        if skill_order == list(skill)[:len(skill_order)]:
            possible_skill_tree_num += 1
            
    return possible_skill_tree_num

File: 프로그래머스/Level 2/스킬트리-ex.java
class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = -1;
        int possibleSkillTreeNum = 0;
        
        for (String skill_tree : skill_trees) {
            if (isPossibleSkillTree(skill, skill_tree)) {
                possibleSkillTreeNum++;
            }
        }
        
        return possibleSkillTreeNum; 
    }
    
    private boolean isPossibleSkillTree(String skill, String skill_tree) {
        int skillIdx = 0;
        
        for (char skillToLearn : skill_tree.toCharArray()) {
            if (skill.indexOf(skillToLearn) != -1) {
                if (skill.charAt(skillIdx) != skillToLearn) {
                    return false; 
                }
                skillIdx++;
            }
        }
        
        return true; 
    }
}